### DISCUSSION TITLE
Auto Build: Run uses incorrent recipe

### DISCUSSION THREAD

**Discussion by spacemonqi** [James-Yu/LaTeX-Workshop#3271] [Q&A]
## Auto Build: Run states that:

When the extension shall auto build LaTeX project using the default (first) recipe.
`onSave` builds the project upon saving a tex file in vscode.
`onFileChange` builds the project upon detecting a file change in any of the dependencies, even modified by other applications.

However, even when I remove all recipes except for the following:

"latex-workshop.latex.recipes": [
        {
            "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
            "tools": [
                "pdflatex",
                "bibtex",
                "pdflatex",
                "pdflatex"
            ]
        },
    ]

Saving / Changing the file still runs pdflatex without bibtex, resulting in citations not being loaded.

Any help would be much appreciated.

**Comment by jlelong** [comment]
Please post the complete logs of the extension starting from `Initializing LaTeX Workshop`. 

![74528024-87b2ef80-4f6a-11ea-9c5d-4f9feff25524](https://user-images.githubusercontent.com/2910140/77959270-98c77e00-72d6-11ea-8c58-4934a9fde39a.png)
